src/constrained_extensions.c

```diff
@@ -0,0 +1,231 @@
+#include "utils.h"
+
+#if PG13_GTE
+
+#include <postgres.h>
+#include <utils/json.h>
+#include <utils/jsonb.h>
+#include <utils/jsonfuncs.h>
+#include <common/jsonapi.h>
+#include <utils/memutils.h>
+#include <miscadmin.h>
+#include <tsearch/ts_locale.h>
+#include <utils/builtins.h>
+#include <utils/memutils.h>
+
+#include <sys/sysinfo.h>
+#include <sys/statvfs.h>
+#include <errno.h>
+
+#include "constrained_extensions.h"
+
+static void
+json_array_start(void *state)
+{
+	json_constrained_extension_parse_state *parse = state;
+
+	parse->state = JCE_UNEXPECTED_ARRAY;
+	parse->error_msg = "unexpected array";
+}
+
+static void
+json_object_start(void *state)
+{
+	json_constrained_extension_parse_state *parse = state;
+
+	switch (parse->state)
+	{
+		case JCE_EXPECT_TOPLEVEL_START:
+			parse->state = JCE_EXPECT_TOPLEVEL_FIELD;
+			break;
+		case JCE_EXPECT_CPU:
+		case JCE_EXPECT_MEM:
+		case JCE_EXPECT_DISK:
+			parse->error_msg = "unexpected object for cpu, mem or disk, expected a value";
```

Comment:
I've tested these parse errors manually. Our test suite doesn't have a way to run with different configs for now.